app/models/health_care_application.rb

```diff
@@ -117,7 +117,7 @@ def self.enrollment_status(icn, loa3)
 
   def self.user_icn(user_attributes)
     HCA::RateLimitedSearch.create_rate_limited_searches(user_attributes) unless Settings.mvi_hca.skip_rate_limit
-    MPI::AttrService.new.find_profile(user_attributes)&.profile&.icn
+    MPI::Service.new.find_profile(user_attributes)&.profile&.icn
```

Comment:
Shouldn't this be `VeteranConfirmation::Service` instead? `MPI::AttrService` had the overridden `create_profile_message` and `measure_info` methods, which have been moved into `VC::Service`.

Reply:
Oh yeah, good catch. Actually I think we should just remove the `VeteranConfirmation::Service` file. The validation that was failing before in `lib/mpi/service.rb` should work now with these changes.

I think `measure_info` will have issues as , but it can be updated to work fine with this use case (it probably was failing because it calls `.uuid` and `.authn_context` on the input `identity` object but since it's just a log we can contextually drop those fields if they don't exist in the input object

Reply:
Those methods were moved to a `VeteranConfirmation` support file since I was under the impression that they were most relevant to `VeteranConfirmation::AttrService.` That's where most of the issues around validations where originating. The `VeteranConfirmation::Service` file was being used in `veteran_confirmation/status_service` to avoid those validation issues.

It does make the whole thing messy and confusing though so I removed `VeteranConfirmation::Service` entirely. The validation still fails however as `MPI::Service` is generally operating with a complex `User` object and VeteranConfirmation is building a very simple OpenStruct.

It seems hacky to check the status of the `user_identity` object and just skip the validations if it's an OpenStruct, but I can't think of another way to handle it that doesn't involve dramatically rewriting a lot of validations. Since the validations were being skipped entirely before anyway I figure there's no reason to do that work now.

Does that make sense?